src/main/kotlin/me/shedaniel/linkie/namespaces/QuiltMappingsNamespace.kt

```diff
@@ -0,0 +1,210 @@
+package me.shedaniel.linkie.namespaces
+
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.builtins.ListSerializer
+import me.shedaniel.linkie.*
+import me.shedaniel.linkie.utils.*
+import org.dom4j.io.SAXReader
+import java.io.InputStream
+import java.net.URL
+import kotlin.collections.component1
+import kotlin.collections.component2
+import kotlin.collections.set
+
+object QuiltMappingsNamespace : Namespace("quilt-mappings") {
+	@Serializable
+	data class QuiltMappingsBuild(
+			val maven: String,
+	)
+
+	val quiltMappingsBuilds = mutableMapOf<String, QuiltMappingsBuild>()
+	val latestQuiltMappingsVersion: String?
+		get() = quiltMappingsBuilds.keys.stream().map { it.tryToVersion() }.filter { it.toString().last().isDigit() && !it.toString().contains("-") }.sorted { a, b -> b!!.compareTo(a!!) }.findFirst().map { it.toString() }.get()
+
+	init {
+		buildSupplier {
+			cached()
+
+			buildVersions {
+				versions { quiltMappingsBuilds.keys }
+				uuid { version ->
+					quiltMappingsBuilds[version]!!.maven.let { it.substring(it.lastIndexOf(':') + 1) }
+				}
+				mappings {
+					MappingsContainer(it, name = "QuiltMappings").apply {
+						loadIntermediaryFromMaven(version)
+						val quiltMappingsMaven = quiltMappingsBuilds[version]!!.maven
+						mappingsSource = loadNamedFromMaven(quiltMappingsMaven.substring(quiltMappingsMaven.lastIndexOf(':') + 1), showError = false)
+					}
+				}
+			}
+		}
+	}
+
+	override fun getDefaultLoadedVersions(): List<String> {
+		return latestQuiltMappingsVersion?.let(::listOf) ?: listOf()
+	}
+
+	override fun getAllVersions(): Sequence<String> = quiltMappingsBuilds.keys.asSequence()
+
+	override fun supportsMixin(): Boolean = true
+	override fun supportsAW(): Boolean = true
+
+	override suspend fun reloadData() {
+		val buildMap = LinkedHashMap<String, MutableList<QuiltMappingsBuild>>()
+
+		val pom = URL("https://maven.quiltmc.org/repository/release/org/quiltmc/quilt-mappings/maven-metadata.xml").readText()
+		SAXReader().read(pom.reader()).rootElement
+				.element("versioning")
+				.element("versions")
+				.elementIterator("version")
+				.asSequence()
+				.map { "org.quitlmc:quilt-mappings:${it.text}" }
+				.forEach {
+					buildMap.computeIfAbsent(it.substring(it.lastIndexOf(":") + 1, it.lastIndexOf("+"))) {
```

Comment:
this piece of code seems confusing, why format it the maven notation way when you are just going to do this the next line?

probably best to format it later when you need it?